### PR TITLE
Updating xcode + detox

### PR DESCRIPTION
**Pull request by chrisdugne**
updated xcode to v10.2, detox to v12

### CODE REVIEW

**Comment by VincentCATILLON** [comment]
Add missing locks but LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -226,7 +226,7 @@ jobs:
             - android/gems
 
       - restore_cache:
-          key: gradle-v1-{{ checksum "android/app/gradle/dependency-locks/releaseCompileClasspath.lockfile" }}-{{ arch }}
+          key: gradle-v2-{{ checksum "android/app/gradle/dependency-locks/releaseCompileClasspath.lockfile" }}-{{ arch }}
```
Comment (VincentCATILLON) [inline .circleci/config.yml]: `gradle-v2-{{ checksum "android/app/gradle/dependency-locks/releaseCompileClasspath.lockfile" }}-{{ checksum "android/app/gradle/dependency-locks/debugAndroidTestCompileClasspath.lockfile" }}-{{ arch }}`

File: .circleci/config.yml [hunk 2]
```diff
@@ -240,15 +240,15 @@ jobs:
             ANDROID_HOME: /opt/android/sdk
 
       - save_cache:
-          key: gradle-v1-{{ checksum "android/app/gradle/dependency-locks/releaseCompileClasspath.lockfile" }}-{{ arch }}
+          key: gradle-v2-{{ checksum "android/app/gradle/dependency-locks/releaseCompileClasspath.lockfile" }}-{{ arch }}
```
Comment (VincentCATILLON) [inline .circleci/config.yml]: don't forget to change it there too

**Comment by VincentCATILLON** [comment]
Guys, do not merge it until this is closed: https://discuss.circleci.com/t/xcode-10-2-support-coomand-line-tools/29417/3

Currently, the Xcode version embedded in 10.2 built image, is a beta version of Xcode (not the GA version). Please follow this [link](https://circleci.com/docs/2.0/testing-ios/) to see when the build of `10.2` version is `10E125` instead of `10P99q` (beta 3).

File: .github/INSTALLATION.md [hunk 3]
```diff
@@ -32,25 +32,14 @@ brew install watchman
   xcode-select --install
   ```
 
-* Fastlane, the mobile toolbox
+* Bundler
+  Bundler provides a consistent environment for Ruby projects by tracking and installing the exact gems and versions that are needed.
```
Comment (Fausto95) [inline .github/INSTALLATION.md]: point to bundler docs pls

**Pull request merged**